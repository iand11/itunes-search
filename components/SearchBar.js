import React from 'react';
import { StyleSheet, TextInput } from 'react-native';

export function SearchBar(props) {
  const { value, onChangeText } = props;
  return (
    <TextInput
      style={styles.textInput}
      placeholder="Search iTunes"
      onChangeText={textInput => onChangeText(textInput)}
      value={value}
    />
  )
}

const styles = StyleSheet.create({
  textInput: {
    height: 40,
    width: '75%',
    backgroundColor: 'white',
    borderWidth: 1,
    borderRadius: 8,
    paddingLeft: 6,
    borderColor: '#ddd'
  }
});