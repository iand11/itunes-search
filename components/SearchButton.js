import React from 'react';
import { StyleSheet, TouchableOpacity, Text } from 'react-native';

export function SearchButton(props) {
  const { handleSearchPress } = props;
  return (
    <TouchableOpacity style={styles.button} onPress={handleSearchPress}>
      <Text style={styles.buttonText}>Search</Text>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  button: {
    height: 40,
    alignItems: 'center',
    backgroundColor: '#aed8e0',
    borderRadius: 8,
    padding: 10
  },
  buttonText: {
    color: 'white'
  }
});