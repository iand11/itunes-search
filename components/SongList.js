import React from 'react';
import { View, Text, FlatList } from 'react-native';

export function SongList(props) {
  renderSongList = () => {
    const { songs, renderSong, height } = props;

    if (!songs) {
      return (
        <View style={{ height: height * .8, justifyContent: 'center', alignItems: 'center' }}>
          <Text style={{ color: 'grey' }}>Type something into the search bar to begin.</Text>
        </View>
      )
    }

    return (
      <FlatList
        data={songs}
        renderItem={renderSong}
        keyExtractor={(item, index) => `${item.song}-${index}`}
      />
    )
  }

  return (
    renderSongList()
  )
}