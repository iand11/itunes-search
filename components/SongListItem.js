import React from 'react';
import { View, TouchableOpacity, Text,  Image, StyleSheet } from 'react-native';
import { MaterialIcons } from '@expo/vector-icons';

export function SongListItem(props) {
  const { index, handleSharePress, handlePlayPress, song, artist, image, preview, songPlaying } = props;

  return (
    <View style={{ flexDirection: 'row', justifyContent: 'space-between', padding: 10 }}>
      <View style={{ width: '60%', flexDirection: 'row', justifyContent: 'flex-start' }}>
        <Image style={{ height: 60, width: 60 }} source={{ uri: image }} />
        <View style={{ flexDirection: 'column', paddingLeft: 10 }}>
          <Text style={styles.songTitleText}>{song}</Text>
          <Text style={styles.artistNameText}>{`by ${artist}`}</Text>
        </View>
      </View>
      <View style={{ width: '20%', justifyContent: 'space-between', flexDirection: 'row', alignItems: 'center' }}>
        <TouchableOpacity onPress={() => handleSharePress({ song, artist, preview })}>
          <MaterialIcons name="share" size={20} color="#aed8e0" />
        </TouchableOpacity>
        <TouchableOpacity onPress={() => handlePlayPress({ song, artist, preview, index })}>
          <MaterialIcons name={songPlaying === `${song}-${index}` ? 'stop' : 'play-circle-filled'} size={40} color="#aed8e0" />
        </TouchableOpacity>
      </View>
    </View>
  )
}

const styles= StyleSheet.create({
  songTitleText: {
    fontSize: 18,
    fontWeight: '500'
  },
  artistNameText: {
    color: 'grey'
  }
})