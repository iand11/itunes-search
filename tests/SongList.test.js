import React from 'react';
import {shallow} from 'enzyme';

import { SongList } from '../components/SongList';
import fetchSongsMock from './mocks/fetchSongsMock';


describe('SongList', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(<SongList songs={null} height={400} renderSong={() => {}}/>);
      expect(component).toMatchSnapshot()
    })
  })

  describe('Interactions', () => {
    it('should not display a list when song prop is null', () => {
      const component = shallow(<SongList songs={null} height={400} renderSong={() => {}}/>);
      const instructions = "Type something into the search bar to begin.";
      const text = component.find('Text').props().children;

      expect(text).toEqual(instructions);
    })

    it('should display a list when song prop is not null', () => {
      const mockLength = fetchSongsMock.length;
      const mockRenderSongs = jest.fn();
      const component = shallow(<SongList songs={fetchSongsMock} height={900} renderSong={mockRenderSongs}/>);
      const list = component.find('FlatList')

      expect(list.length).toEqual(1);
      expect(list.props().data.length).toEqual(mockLength);
    })
  })
})