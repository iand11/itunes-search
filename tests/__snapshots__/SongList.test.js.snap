// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SongList Rendering should match to snapshot 1`] = `ShallowWrapper {}`;
